xbmc/PlayListPlayer.cpp

```diff
@@ -258,13 +258,23 @@ bool CPlayListPlayer::PlaySongId(int songId)
 bool CPlayListPlayer::Play(const CFileItemPtr& pItem, const std::string& player)
 {
   int playlist;
-  if (pItem->IsAudio())
+  bool isVideo{pItem->IsVideo()};
+  bool isAudio{pItem->IsAudio()};
+  if (isVideo && isAudio && pItem->HasProperty("playlist_type_hint"))
+  {
+    // If an extension is set in both audio / video lists (e.g. playlist .strm),
+    // is not possible detect the type of playlist then we rely on the hint
+    playlist = pItem->GetProperty("playlist_type_hint").asInteger32();
```

Comment:
probably this needs to be validated before calling `.asInteger32()`? I.e. plugin authors may set that property to some random string. What happens in that case?
I've not checked the implementation of the `.asInteger32()` though so it might be fine already. Please double check if possible.

Reply:
i checked and yes its safe, so if the property is set with data other than integer, then the method fallback to default value,
so i have add the default value to PLAYLIST_NONE

Reply:
I'd say that if a plugin resolves its url to a local or remote playlist (e.g. strm) yes, the property will still be set from python although the dynpath changed